Chess in Python.py

```diff
@@ -1,33 +1,66 @@
 from __future__ import annotations
 from enum import Enum, auto
+import re
 
 # Global Variables
 emptyCell = ''
 
-# global utlity function to check if the user input is an int between 1 and 8. 
-# to be used when validating user inputs of board positions for selecting a piece they want to move.
-def askFor1To8(message) -> int:  # get input from user with message
-    userInput = input(message) # check if userInput is a string; pass if the ValueError is thrown
-    try:
-        userInput = int(userInput)
-    except ValueError:
-        userInput = -1
-        pass
-    if userInput <= 8 and userInput > 0: # validate that the userInput is between 1 and 8
-        return(userInput)
-    return askFor1To8("Try again ") # recursively return a call of the function if the user input is invalid
-
-# global utility function for checking if a piece move is valid
-def askForValidPiece(message) -> Piece:
+# Regular Expression for valid moves:
+#   (letter-from-a-to-h) (number-from-1-to-8) hyphen (letter-from-a-to-h) (number-from-1-to-8)
+# (the parentheses around each letter and number capture it in a capture group)
+# So, if there's a match, the capture groups will be:
+#   source_file = match.group(1)
+#   source_rank = match.group(2)
+#   dest_file = match.group(3)
+#   dest_rank = match.group(4)
+longNotationPattern = "^([a-h])([1-8])-([a-h])([1-8])$"
+
+
+class SquareLocation:
+    file: int
+    rank: int
+    file_codes = ["a", "b", "c", "d", "e", "f", "g", "h"]
+
+    def __init__(self, file, rank):
+        self.file = ord(file) - ord("a")
+        self.rank = int(rank) - 1
+
+    def __str__(self):
+        return f"{str(self.file_codes[self.file])}{str(self.rank + 1)}"
```

Comment:
@mknowles36 question: why do you initialize self.rank (line 26) as being the rank minus one, only to add one to self.rank when you're printing self.rank (line 29)?

Reply:
Right - so the idea was that I wanted the internally-stored rank and file in SquareLocation to be in a standard index-of-array form (ie., 0-based).  So that's why I subtracted ord('a') from the file and 1 from the rank.  From the standard form, it can then be converted to either back to Algebraic Notation form (in __str__()) or to X/Y coordinates of the storage array (in getXY()) via the transforms in those corresponding methods (ie adding 1 back to the rank and effectively adding 'a' back to the file (via a lookup table).

Having said that, you do make a point that storing the rank and file in the standard 0-based form requires some form of transform to use in both getXY() and __str__() and is therefore somewhat less-than-optimal.  A more optimal way would be to either just store the rank and file as given in __init__() and then just do the appropriate conversion in getXY(), or convert it to array X-Y coords in __init__() (which would make getXY() trivial) and do the conversion back to Algebraic Notation in __str__().    Of those two approaches, we should choose the one that allows for the trivial implementation the most-used of those two methods (getXY() vs. __str__()).  Want me to add a commit with that change?